repo/migrations/Migration027.go

```diff
@@ -1,683 +1,551 @@
+		CryptoCurrencyCode string                       `json:"coinType"`
+	}
+
+	Migration027V4price struct {
+		CurrencyCode string  `json:"currencyCode"`
+		Amount       uint    `json:"amount"`
+		Modifier     float32 `json:"modifier"`
+	}
+
+	Migration027V4ListingIndexData struct {
+		Hash               string                       `json:"hash"`
+		Slug               string                       `json:"slug"`
+		Title              string                       `json:"title"`
+		Categories         []string                     `json:"categories"`
+		NSFW               bool                         `json:"nsfw"`
+		ContractType       string                       `json:"contractType"`
+		Description        string                       `json:"description"`
+		Thumbnail          Migration027ListingThumbnail `json:"thumbnail"`
+		Price              Migration027V4price          `json:"price"`
+		ShipsTo            []string                     `json:"shipsTo"`
+		FreeShipping       []string                     `json:"freeShipping"`
+		Language           string                       `json:"language"`
+		AverageRating      float32                      `json:"averageRating"`
+		RatingCount        uint32                       `json:"ratingCount"`
+		ModeratorIDs       []string                     `json:"moderators"`
+		AcceptedCurrencies []string                     `json:"acceptedCurrencies"`
+		CryptoCurrencyCode string                       `json:"coinType"`
+	}
+)
+
+var divisibilityMap = map[string]uint{
+	"BTC": 8,
+	"BCH": 8,
+	"LTC": 8,
+	"ZEC": 8,
+}
+
+func parseV5intoV4(v5 Migration027V5ListingIndexData) Migration027V4ListingIndexData {
+	return Migration027V4ListingIndexData{
+		Hash:         v5.Hash,
+		Slug:         v5.Slug,
+		Title:        v5.Title,
+		Categories:   v5.Categories,
+		NSFW:         v5.NSFW,
+		ContractType: v5.ContractType,
+		Description:  v5.Description,
+		Thumbnail:    v5.Thumbnail,
+		Price: Migration027V4price{
+			CurrencyCode: string(v5.Price.Currency.Code),
+			Amount:       uint(v5.Price.Amount.Uint64()),
+			Modifier:     v5.Modifier,
+		},
+		ShipsTo:            v5.ShipsTo,
+		FreeShipping:       v5.FreeShipping,
+		Language:           v5.Language,
+		AverageRating:      v5.AverageRating,
+		RatingCount:        v5.RatingCount,
+		ModeratorIDs:       v5.ModeratorIDs,
+		AcceptedCurrencies: v5.AcceptedCurrencies,
+		CryptoCurrencyCode: v5.CryptoCurrencyCode,
+	}
+}
+
+func parseV4intoV5(v4 Migration027V4ListingIndexData) Migration027V5ListingIndexData {
+	var priceValue *Migration027V5CurrencyValue
+	if v4.Price.CurrencyCode != "" {
+		divisibility, ok := divisibilityMap[strings.ToUpper(v4.Price.CurrencyCode)]
+		if !ok {
+			divisibility = 2
+		}
+
+		priceValue = &Migration027V5CurrencyValue{
+			Amount: new(big.Int).SetInt64(int64(v4.Price.Amount)),
+			Currency: Migration027V5CurrencyDefinition{
+				Code:         Migration027V5CurrencyCode(v4.Price.CurrencyCode),
+				Divisibility: divisibility,
+			},
+		}
+	}
```

Comment:
This is going to leave empty objects when v`v4.Price.CurrencyCode` is empty. Have you confirmed that this works when consumed?